ethereumj-core/src/main/java/org/ethereum/crypto/ECKey.java

```diff
@@ -345,9 +516,13 @@ public ECPoint getPubKeyPoint() {
      * @throws java.lang.IllegalStateException if the private key bytes are not available.
      */
     public BigInteger getPrivKey() {
-        if (priv == null)
+        if (privKey == null) {
+            throw new MissingPrivateKeyException();
+        } else if (privKey instanceof BCECPrivateKey) {
+            return ((BCECPrivateKey) privKey).getS();
```

Comment:
Oh, I see your original point through…
I used a mix of `.getS()` and `.getD()` in various methods.
No, that wasn’t a conscious choice. I can certainly clean that up to be consistent.